app/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/ExciseNumberQueueWorkItemRepository.scala

```diff
@@ -36,19 +40,48 @@ class ExciseNumberQueueWorkItemRepository @Inject()
   collectionName = "excise-number-work-item",
   mongoComponent = mongoComponent,
   itemFormat = ExciseNumberWorkItem.format,
-  workItemFields = WorkItemFields.default,
+  workItemFields = WorkItemFields.default.copy(availableAt = "availableAt", receivedAt = "lastSubmitted"),
   extraIndexes = Seq(
     IndexModel(
-      Indexes.ascending("updatedAt"),
-      IndexOptions().expireAfter(appConfig.getMovementTTL.toSeconds, TimeUnit.SECONDS)
+      Indexes.ascending("lastSubmitted"),
+      IndexOptions().expireAfter(appConfig.workItemTTL.toSeconds, TimeUnit.SECONDS)
+    ),
+    IndexModel(
+      Indexes.ascending("item.exciseNumber"),
+      IndexOptions()
+        .name("item_ern_index")
+        .unique(true)
     )
   )
 ) {
 
   override def now(): Instant = timeService.timestamp()
 
   override val inProgressRetryAfter: Duration = {
-    appConfig.retryAfterMinutes
+    appConfig.workItemInProgressTimeOut.toJava
+  }
+
+  def getWorkItemForErn(ern: String): Future[Seq[WorkItem[ExciseNumberWorkItem]]] = {
+    //Returns a Seq but should only ever be one or zero, as "item.exciseNumber" is a unique index
+
+    collection
+      .find(
+        in("item.exciseNumber", ern),
+      )
+      .toFuture()
+  }
+
+  def saveUpdatedWorkItem(updatedWI: WorkItem[ExciseNumberWorkItem]): Future[Boolean] = {
+    val update = combine(
+      set("item", Codecs.toBson(updatedWI.item)),
+      set("availableAt", updatedWI.availableAt),
```

Comment:
just a curiosity, should this be transformed to toBison?

Reply:
It didn't work with toBson and does without